### PR TITLE
Upgrade WizNote.app to 0.1.66

### PR DESCRIPTION
**Pull request by aaronjheng**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making all changes to a cask, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [ ] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [ ] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [ ] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [ ] `brew audit --new-cask <cask>` worked successfully.
- [ ] `brew install --cask <cask>` worked successfully.
- [ ] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

File: Casks/wiznote.rb [hunk 1]
```diff
@@ -1,15 +1,36 @@
 cask "wiznote" do
-  version "2020-08-20"
-  sha256 "043c8bf25233d711bbeb2354808dae79c431373cd9989ccdeb46ed8292b1db55"
```
Comment (neutric) [inline Casks/wiznote.rb:3]: `arch = Hardware::CPU.intel? ? "x64" : "arm64"`

File: Casks/wiznote.rb [hunk 2]
```diff
@@ -1,15 +1,36 @@
...
+  version "0.1.66"
+  sha256 :no_check
 
-  url "https://get.wiz.cn/wiznote-macos-#{version}.dmg"
+  if Hardware::CPU.intel?
+    url "https://url.wiz.cn/u/mac64_new"
+  else
+    url "https://url.wiz.cn/u/mac_arm_new"
+  end
```
Comment (neutric) [inline Casks/wiznote.rb:9]: `url "https://wizteam-generic.pkg.coding.net/client/wiznote-autoupdate/wiznote-desktop-#{version}-mac-#{arch}.dmg?version=latest"` and then test if you can leave `?version=latest` out. The versioned `url` allows for checksums, so change `sha256` accordingly.

File: Casks/wiznote.rb [hunk 3]
```diff
@@ -1,15 +1,36 @@
...
   name "WizNote"
-  homepage "https://www.wiz.cn/wiznote-mac.html"
+  desc "For note-taking"
```
Comment (neutric) [inline Casks/wiznote.rb:11]: `desc "Note-taking application"`

File: Casks/wiznote.rb [hunk 4]
```diff
@@ -1,15 +1,36 @@
...
+  homepage "https://www.wiz.cn/"
 
   livecheck do
-    url "https://url.wiz.cn/u/mac"
+    url :url
     strategy :header_match
+    regex(/wiznote-desktop-(\d+(?:\.\d+)*)-mac-x64.dmg/i)
```
Comment (neutric) [inline Casks/wiznote.rb:17]: `url "https://url.wiz.cn/u/mac64_new"`. Or the one for arm, probably doesn't matter. Specify a regex only if the implicit, default regex doesn't pick up the `version` correctly.

File: Casks/wiznote.rb [hunk 5]
```diff
@@ -1,15 +1,36 @@
...
   end
 
+  auto_updates true
+
+  depends_on macos: ">= :sierra" if Hardware::CPU.intel?
```
Comment (neutric) [inline Casks/wiznote.rb:22]: `if Hardware::CPU.intel?` can be deleted.

File: Casks/wiznote.rb [hunk 6]
```diff
@@ -1,15 +1,36 @@
...
+
+  zap trash: [
+    "~/Library/Caches/cn.wiznote.desktop",
+    "~/Library/Caches/cn.wiznote.desktop.ShipIt",
+    "~/Library/Logs/WizNote",
+    "~/Library/Preferences/ByHost/cn.wiznote.desktop.ShipIt.*.plist",
+    "~/Library/Preferences/cn.wiznote.desktop.plist",
+    "~/Library/Saved Application State/cn.wiznote.desktop.savedState",
+    "~/Library/Application Support/Caches/wiznote-desktop-updater",
+    "~/Library/Application Support/WizNote",
```
Comment (neutric) [inline Casks/wiznote.rb:34]: Please sort the `zap` entries alphabetically.

**Review by neutric — COMMENTED** [review]
Thanks for your contribution to `homebrew-cask`, @aaronjheng! My review is in the code. Take it with a grain of salt as I am not a maintainer.

File: Casks/wiznote.rb (hunk 4 above)
Comment (neutric) [reply Casks/wiznote.rb:16]: Or, if you want it extra nice, do a conditional assignment for arch like above for the download `url` and than use it in the livecheck `url`.

**Comment by aaronjheng** [comment]
@neutric Thanks for the tips. All great. It means a lot.

File: Casks/wiznote.rb [hunk 7]
```diff
@@ -1,15 +1,34 @@
 cask "wiznote" do
-  version "2020-08-20"
-  sha256 "043c8bf25233d711bbeb2354808dae79c431373cd9989ccdeb46ed8292b1db55"
+  arch = Hardware::CPU.intel? ? "x64" : "arm64"
 
-  url "https://get.wiz.cn/wiznote-macos-#{version}.dmg"
+  version "0.1.66"
+  sha256 "2616f22d08cf8ecaedd8b0133de51bd60b2ad9d9e81d38594a9e99143af504f8"
```
Comment (neutric) [inline Casks/wiznote.rb:5]: The checksum for the 2nd arch is missing.
```
if Hardware::CPU.intel?
  sha256 "x64-checksum here"
else
  sha256 "arm64-checksum here"
end
```

File: Casks/wiznote.rb [hunk 8]
```diff
@@ -1,15 +1,34 @@
...
   name "WizNote"
-  homepage "https://www.wiz.cn/wiznote-mac.html"
+  desc "Note-taking application"
+  homepage "https://www.wiz.cn/"
 
   livecheck do
-    url "https://url.wiz.cn/u/mac"
+    url "https://url.wiz.cn/u/mac64_new"
     strategy :header_match
+    regex(/wiznote-desktop-(\d+(?:\.\d+)*)-mac-x64.dmg/i)
```
Comment (neutric) [inline Casks/wiznote.rb:16]: Specifying a `regex` is most likely not necessary, did you try without?

**Review by neutric — COMMENTED** [review]
Second round. :)

**Comment by miccal** [comment]
Thank you @aaronjheng and @neutric.

**Pull request merged**